File: generators/app/templates/OtherTest.java
package <%= package %>;

import <%= package %>.AppProvider.DateParameterConverter;
import com.google.gson.Gson;
import java.util.Date;
import org.junit.Test;
import static org.junit.Assert.*;

/**
 * Test other services
 * @author martinlabs CRUD generator
 */
public class OtherTest {

    @Test
    public void testErrorCodeInstance() {
        ErrorCode ec = new ErrorCode();
    }
    
    @Test
    public void testAppProvider() {
        AppProvider appProvider = new AppProvider();
        
        Gson context = appProvider.getContext(null);
        DateParameterConverter converter = (DateParameterConverter) appProvider.getConverter(Date.class, null, null);
        
        assertNotNull(context);
        assertNotNull(converter);
    }
    
}

File: generators/app/templates/OtherTest.kt
package <%= package %>

import <%= package %>.AppProvider.DateParameterConverter
import com.google.gson.Gson
import java.util.Date
import org.junit.Test
import org.junit.Assert.*

/**
 * Test other services
 * @author martinlabs CRUD generator
 */
class OtherTest {

    @Test
    fun testAppProvider() {
        val appProvider = AppProvider()

        val context = appProvider.getContext(null)
        val converter = appProvider.getConverter(Date::class.java, null, null) as DateParameterConverter?

        assertNotNull(context)
        assertNotNull(converter)
    }

}
